File: app/lib/features/auth/domain/usecases/register_use_case.dart
import '../entities/auth_session.dart';
import '../repositories/auth_repository.dart';

class RegisterUseCase {
  final AuthRepository repository;

  RegisterUseCase(this.repository);

  Future<AuthSession> call({
    required String firstName,
    required String lastName,
    required String email,
    required String password,
    required String role,
    String? gymId,
    String? gymName,
    String? gymSlug,
    String? phone,
  }) {
    return repository.register(
      firstName: firstName,
      lastName: lastName,
      email: email,
      password: password,
      role: role,
      gymId: gymId,
      gymName: gymName,
      gymSlug: gymSlug,
      phone: phone,
    );
  }
}

File: panel/src/features/auth/domain/use-cases/RegisterUseCase.ts
import type { AuthSession } from '../entities/AuthSession';
import { AuthRepository } from '../repositories/AuthRepository';

export class RegisterUseCase {
  constructor(private readonly repository: AuthRepository) {}

  execute(input: {
    firstName: string;
    lastName: string;
    email: string;
    password: string;
    role: string;
    phone?: string;
    gymId?: string;
    gymName?: string;
    gymSlug?: string;
  }): Promise<AuthSession> {
    return this.repository.register(input);
  }
}